### PR TITLE
Clarification about OAuth

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -861,7 +861,7 @@ The term "Subject" provides an identifier of the Issuer's choosing to refer to a
...
 
-In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier", and then later as an "Issuer". This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
+In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens. This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service. Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:864]:
```suggestion
In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -861,7 +861,9 @@ The term "Subject" provides an identifier of the Issuer's choosing to refer to a
 
 In simpler language, a SCITT Statement could be some vendor-specific software bill of materials (SBOM), results from a model checker, static analyzer, or RATS Evidence about the authenticity of an SBOM creation process, where the Issuer identifies themselves using the `iss` Claim, and the specific software that was analyzed as the Subject using the `sub` Claim.
 
-In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier", and then later as an "Issuer". This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
+In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:864]:
```suggestion
In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer".
This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
```

**Review by OR13 — APPROVED** [review]

**Review by roywill — APPROVED** [review]

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -861,7 +861,10 @@ The term "Subject" provides an identifier of the Issuer's choosing to refer to a
...
 
-In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier", and then later as an "Issuer". This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
+In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer".
+This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
+This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
+Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:867]:
```suggestion
Highlighting {{RFC7523}} is intended as an example for client authentication originating from OAuth. Many other approaches are applicable.
```

**Review by henkbirkholz — CHANGES_REQUESTED** [review]
Still using the term Verifier... but it is in quotes and used as an analogy. good enough, I hope.

Added a suggested change that emphasizes that the OAuth domain is really just an example.

File: draft-ietf-scitt-architecture.md (hunk 3 above)
Comment (hannestschofenig) [reply draft-ietf-scitt-architecture.md:867]: "Many other approaches are applicable." --> The reader will ask himself/herself "which other approaches"

Comment (hannestschofenig) [reply draft-ietf-scitt-architecture.md:867]: The example of OAuth is actually not so useful as you think. The JWT that is used for client authentication is really only valuable if the JWT has previously been obtained in a secure way. Just creating the JWT on the fly does not provide any useful authentication -- for obvious reason. 

In that sense, the JWT for client authentication in OAuth should be seen as a substitute for a X.509 certificate

**Review by henkbirkholz — APPROVED** [review]

**Pull request merged**